### PR TITLE
Added remainder for primitive arrays

### LINKED ISSUE #128: Add modulo to compute
https://en.wikipedia.org/wiki/Modulo_operation

### PR DESCRIPTION
**Pull request by jorgecarleitao**
Closes #128

### CODE REVIEW

File: src/compute/arithmetics/basic/rem.rs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    array::{Array, PrimitiveArray},
+    compute::{
+        arithmetics::{ArrayCheckedRem, ArrayRem, NotI128},
+        arity::{binary, binary_checked, unary, unary_checked},
+    },
+    error::{ArrowError, Result},
+    types::NativeType,
+};
+
+/// Reminder of two primitive arrays with the same type.
```
Comment (Dandandan) [inline src/compute/arithmetics/basic/rem.rs:15]:
```suggestion
/// Remainder of two primitive arrays with the same type.
```

File: src/compute/arithmetics/basic/rem.rs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    if lhs.data_type() != rhs.data_type() {
+        return Err(ArrowError::InvalidArgumentError(
+            "Arrays must have the same logical type".to_string(),
+        ));
+    }
+
+    binary(lhs, rhs, lhs.data_type().clone(), |a, b| a % b)
+}
+
+/// Checked reminder of two primitive arrays. If the result from the reminder
```
Comment (Dandandan) [inline src/compute/arithmetics/basic/rem.rs:42]:
```suggestion
/// Checked remainder of two primitive arrays. If the result from the remainder
```

File: src/compute/arithmetics/basic/rem.rs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    T: NativeType + CheckedRem<Output = T> + Zero + NotI128,
+{
+    type Output = Self;
+
+    fn checked_rem(&self, rhs: &PrimitiveArray<T>) -> Result<Self::Output> {
+        checked_rem(self, rhs)
+    }
+}
+
+/// Reminder a primitive array of type T by a scalar T.
```
Comment (Dandandan) [inline src/compute/arithmetics/basic/rem.rs:94]:
```suggestion
/// Remainder a primitive array of type T by a scalar T.
```

File: src/compute/arithmetics/basic/rem.rs [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+/// ```
+pub fn rem_scalar<T>(lhs: &PrimitiveArray<T>, rhs: &T) -> PrimitiveArray<T>
+where
+    T: NativeType + Rem<Output = T>,
+{
+    let rhs = *rhs;
+    unary(lhs, |a| a % rhs, lhs.data_type().clone())
+}
+
+/// Checked reminder of a primitive array of type T by a scalar T. If the
```
Comment (Dandandan) [inline src/compute/arithmetics/basic/rem.rs:115]:
```suggestion
/// Checked remainder of a primitive array of type T by a scalar T. If the
```

File: src/compute/arithmetics/mod.rs [hunk 5]
```diff
@@ -113,6 +113,12 @@ pub fn arithmetic(lhs: &dyn Array, op: Operator, rhs: &dyn Array) -> Result<Box<
                 Subtract => decimal::sub::sub(lhs, rhs),
                 Multiply => decimal::mul::mul(lhs, rhs),
                 Divide => decimal::div::div(lhs, rhs),
+                Reminder => {
```
Comment (Dandandan) [inline src/compute/arithmetics/mod.rs:116]:
```suggestion
                Remainder => {
```

File: src/compute/arithmetics/mod.rs [hunk 6]
```diff
@@ -214,6 +224,7 @@ pub enum Operator {
     Subtract,
     Multiply,
     Divide,
+    Reminder,
```
Comment (Dandandan) [inline src/compute/arithmetics/mod.rs:227]:
```suggestion
    Remainder,
```

**Review by elferherrera — COMMENTED** [review]
You read my mind. I was adding this operation to Nushell and I was thinking that it is missing in Arrow2. The implementation looks good as usual

**Pull request merged**